tests/nnstreamer_source/unittest_src_iio.cpp

```diff
@@ -1434,6 +1440,36 @@ TEST (test_tensor_src_iio, data_verify_freq_generic_type)
   EXPECT_EQ (status, GST_STATE_CHANGE_SUCCESS);
   EXPECT_EQ (state, GST_STATE_PLAYING);
 
+  /** get and verify the caps */
+  src_iio = gst_bin_get_by_name (GST_BIN (src_iio_pipeline), "my-src-iio");
+  ASSERT_NE (src_iio, nullptr);
+  src_pad = gst_element_get_static_pad (src_iio, "src");
+  ASSERT_NE (src_pad, nullptr);
+  caps = gst_pad_get_current_caps (src_pad);
+  ASSERT_NE (caps, nullptr);
+  structure = gst_caps_get_structure (caps, 0);
+  ASSERT_NE (structure, nullptr);
+
+  /** Default has merge channels enabled */
+  EXPECT_STREQ (gst_structure_get_name (structure), "other/tensors");
+  EXPECT_EQ (gst_tensors_config_from_structure (&config, structure), TRUE);
+  EXPECT_EQ (config.rate_n, samp_freq);
```

Comment:
I wonder that gst_object_unref (src_iio_pipeline) guarantee us unreffering those src_pad, caps, and structure included in src_iio_pipleine. If not, could we do get_object_unref() for variables, src_pad, caps, and structure, here? It seems that there are no code using those variables  after this line. Near the line for gst_object_unref (src_iio_pipeline) looks also good to me.

Reply:
Yes, gst_object_unref (src_iio_pipeline) would only free its resources along with added elements in the pipeline, if they are referenced. Added unref for bin, pad, caps.
Structure need not be unref as per documentation.